pkg/dynamicinstrumentation/ebpf/ebpf.go

```diff
@@ -54,7 +53,43 @@ func AttachBPFUprobe(procInfo *ditypes.ProcessInfo, probe *ditypes.Probe) error
 		return fmt.Errorf("could not create bpf collection for probe %s: %w", probe.ID, err)
 	}
 
-	mapReplacements := map[string]*ebpf.Map{}
+	inner := &ebpf.MapSpec{
+		Type:       ebpf.Stack,
+		MaxEntries: 2048,
+		ValueSize:  8,
+	}
+
+	outer := &ebpf.MapSpec{
+		Name:       "param_stacks",
+		InnerMap:   inner,
+		Type:       ebpf.ArrayOfMaps,
+		MaxEntries: uint32(32), // we arbitrarily set this to support 32 CPUs
```

Comment:
You can get the exact value of available CPUs easily:
```
numCPUs, err := kernel.PossibleCPUs()
if err != nil {
	return nil, fmt.Errorf("error getting possible cpus: %s", err)
}
outer.MaxEntries = numCPUs
```

Reply:
Another reason to do this, 32 cpus is not sufficient for even our internal environments. I think we have some nodes with 96 cpus.